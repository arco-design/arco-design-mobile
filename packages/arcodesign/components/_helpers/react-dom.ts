import { ReactElement } from 'react';
import * as ReactDOM from 'react-dom';

function isObject(obj: any): obj is { [key: string]: any } {
    return Object.prototype.toString.call(obj) === '[object Object]';
}

export interface RootType {
    render: (element: ReactElement) => void;
    _unmount: () => void;
}
export interface RootTypeReact extends RootType {
    unmount?: () => void;
}
export type CreateRootFnType = (container: Element | DocumentFragment) => {
    render: (element: ReactElement) => void;
    unmount?: () => void;
};

// Cast ReactDOM to a version that might have createRoot and other properties
const typedReactDOM = ReactDOM as typeof ReactDOM & {
    createRoot?: CreateRootFnType;
    render?: (
        element: ReactElement,
        container: Element | DocumentFragment,
        callback?: () => void,
    ) => any;
    unmountComponentAtNode?: (container: Element | DocumentFragment) => boolean;
    __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED?: {
        usingClientEntryPoint?: boolean;
    };
};

let copyRender: (
    app: ReactElement,
    container: Element | DocumentFragment,
    createRootFunction?: CreateRootFnType,
) => RootType;

const updateUsingClientEntryPoint = (skipWarning?: boolean) => {
    // https://github.com/facebook/react/blob/17806594cc28284fe195f918e8d77de3516848ec/packages/react-dom/npm/client.js#L10
    // Avoid console warning
    const secretInternals = typedReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    if (isObject(secretInternals)) {
        secretInternals.usingClientEntryPoint = skipWarning;
    }
};

const createRootFn = typedReactDOM.createRoot;

const getRender =
    (createRootFunction: CreateRootFnType) =>
    (app: ReactElement, container: Element | DocumentFragment) => {
        updateUsingClientEntryPoint(true);
        const root = createRootFunction!(container);
        updateUsingClientEntryPoint(false);

        root.render(app);

        return {
            render: (elementToRender: ReactElement) => {
                root.render(elementToRender);
            },
            _unmount() {
                setTimeout(() => {
                    if (root && typeof root.unmount === 'function') {
                        root.unmount();
                    }
                });
            },
        };
    };

if (createRootFn) {
    // React 18
    copyRender = getRender(createRootFn);
} else if (
    typeof typedReactDOM.render === 'function' &&
    typeof typedReactDOM.unmountComponentAtNode === 'function'
) {
    // React 16/17
    copyRender = function (app: ReactElement, container: Element | DocumentFragment) {
        typedReactDOM.render!(app, container); // Use non-null assertion
        return {
            render: (elementToRender: ReactElement) => {
                typedReactDOM.render!(elementToRender, container); // Use non-null assertion
            },
            _unmount() {
                typedReactDOM.unmountComponentAtNode!(container); // Use non-null assertion
            },
        };
    };
} else {
    copyRender = (
        app: ReactElement,
        container: Element | DocumentFragment,
        createRootFunction?: CreateRootFnType,
    ) => {
        const defaultCb = () => {
            // Fallback if no rendering method is found
            console.error(
                'ArcoDesign: ReactDOM.createRoot() or ReactDOM.render() not found. ' +
                    'This usually means you are using an unsupported version of React, ' +
                    'or ReactDOM is not properly initialized. ' +
                    'ArcoDesign Mobile React requires React 16, 17, 18, or 19.',
            );
            return { render: (_element: ReactElement) => {}, _unmount: () => {} };
        };
        if (createRootFunction) {
            return getRender(createRootFunction)(app, container);
        }
        return defaultCb();
    };
}

export const render = copyRender;
