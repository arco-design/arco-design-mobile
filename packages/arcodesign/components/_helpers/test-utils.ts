import React, { useEffect, useRef } from 'react';
import { Promise as ES6Promise, Promise } from 'es6-promise';

/**
 * React 19 兼容的 act 包装器
 * 用于包装可能触发状态更新的异步操作
 */
export const actWrapper = (callback: () => void | ES6Promise<void>): ES6Promise<void> => {
    return new ES6Promise<void>((resolve, reject) => {
        try {
            // 尝试使用 react-dom/test-utils 的 act
            const testUtils = require('react-dom/test-utils');
            if (testUtils && typeof testUtils.act === 'function') {
                // 对于同步回调
                if (typeof callback === 'function') {
                    try {
                        const result = callback();
                        if (result && typeof result.then === 'function') {
                            testUtils
                                .act(() => result)
                                .then(() => resolve())
                                .catch(reject);
                        } else {
                            testUtils.act(() => {
                                callback();
                            });
                            resolve();
                        }
                    } catch (error) {
                        reject(error);
                    }
                } else {
                    resolve();
                }
            } else {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            }
        } catch (error) {
            // 如果 act 不可用，直接执行回调
            try {
                const result = callback();
                if (result && typeof result.then === 'function') {
                    result.then(() => resolve()).catch(reject);
                } else {
                    resolve();
                }
            } catch (callbackError) {
                reject(callbackError);
            }
        }
    });
};

/**
 * 清理未处理的异步操作，防止内存泄漏警告
 */
export const cleanupAsync = (): ES6Promise<void> => {
    return new ES6Promise<void>(resolve => {
        setTimeout(() => resolve(), 0);
    });
};

/**
 * 安全的状态更新，检查组件是否已卸载
 */
export const createSafeStateUpdater = (isMountedRef: React.MutableRefObject<boolean>) => {
    return (updater: () => void) => {
        if (isMountedRef.current) {
            updater();
        }
    };
};

/**
 * 用于测试的 useEffect 清理函数
 */
export const useTestCleanup = (cleanupFn: () => void) => {
    const isMountedRef = useRef(true);

    useEffect(() => {
        return () => {
            isMountedRef.current = false;
            cleanupFn();
        };
    }, [cleanupFn]);

    return isMountedRef;
};
