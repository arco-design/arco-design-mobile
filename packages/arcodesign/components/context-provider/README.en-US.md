### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
