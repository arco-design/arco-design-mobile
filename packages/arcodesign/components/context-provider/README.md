### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
