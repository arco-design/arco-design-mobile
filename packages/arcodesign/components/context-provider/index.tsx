import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
