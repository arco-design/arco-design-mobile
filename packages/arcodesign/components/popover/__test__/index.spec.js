import React, { createRef } from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { defaultContext } from '../../context-provider';
import { mockElementProperty } from '../../../tests/helpers/mockElement';
import { eventCommonProps, mockAddListener } from '../../../tests/helpers/mockEvent';
import Popover from '..';
import Button from '../../button';

demoTest('popover');

mountTest(Popover, 'Popover');

const prefix = `${defaultContext.prefixCls}-popover`;

function clickButton(wrapper, event = 'click') {
    wrapper.find('button').simulate(event);
    act(() => {
        jest.advanceTimersByTime(800);
    });
    wrapper.update();
}

function mockPopoverSize(ref, width, height) {
    const { innerPopover } = ref.current;
    expect(typeof innerPopover).toBe('object');
    const { content } = innerPopover || {};
    expect(typeof content).toBe('object');
    mockElementProperty(content, 'offsetWidth', width);
    mockElementProperty(content, 'offsetHeight', height);
}

function checkHasPopoverInner(wrapper) {
    expect(wrapper.find(`.${prefix}-inner`)).toHaveLength(1);
}

function checkHasNoPopoverInner(wrapper) {
    expect(wrapper.find(`.${prefix}-inner`)).toHaveLength(0);
}

function updatePopoverPosition(wrapper, ref) {
    const { updatePosition } = ref.current;
    expect(typeof updatePosition).toBe('function');
    act(() => {
        updatePosition();
    });
    wrapper.update();
}

describe('Popover', () => {
    beforeEach(() => {
        jest.useFakeTimers();
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('should open and close popover correctly', () => {
        const onChange = jest.fn();
        const ref = createRef();
        const wrapper = mount(<Popover
            content="Bubble"
            onChange={onChange}
            showCloseIcon={true}
            textSuffix="suffix"
            clickSelfToClose={false}
            ref={ref}
            direction="topRight"
        >
            <Button>Button</Button>
        </Popover>);
        const map = mockAddListener(document.body, true);
        checkHasNoPopoverInner(wrapper);
        clickButton(wrapper);
        mockPopoverSize(ref, 100, 60);
        updatePopoverPosition(wrapper, ref);
        expect(onChange.mock.calls).toHaveLength(1);
        expect(onChange.mock.calls[0]).toEqual([true]);
        checkHasPopoverInner(wrapper);
        expect(wrapper.find('.popover-content')).toHaveLength(1);
        expect(wrapper.find(`.${prefix}-inner`).hasClass('with-suffix')).toBe(true);
        expect(wrapper.find('.content-text')).toHaveLength(1);
        expect(wrapper.find('.text-close-icon')).toHaveLength(1);
        expect(wrapper.find('.text-suffix')).toHaveLength(1);

        // should not close when clickSelfToClose=false and click button
        clickButton(wrapper);
        expect(onChange.mock.calls).toHaveLength(1);
        checkHasPopoverInner(wrapper);

        // should close when clickSelfToClose=true and click button
        wrapper.setProps({ clickSelfToClose: true, clickOtherToClose: true, preventBodyClick: true });
        clickButton(wrapper);
        expect(onChange.mock.calls).toHaveLength(2);
        expect(onChange.mock.calls[1]).toEqual([false]);
        checkHasNoPopoverInner(wrapper);

        // should close when clickOtherToClose=true and click body
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);
        act(() => {
            map.click({ target: document.body, ...eventCommonProps });
            jest.advanceTimersByTime(800);
        });
        wrapper.update();
        checkHasNoPopoverInner(wrapper);

        // should not close when touchToClose=true and button's click event triggered
        wrapper.setProps({ touchToClose: true });
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);

        // should not close when touchToClose=true and button's touchstart event triggered
        clickButton(wrapper, 'touchstart');
        checkHasNoPopoverInner(wrapper);

        // should support touch body to close
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);
        act(() => {
            map.touchstart({ targetTouches: [{ target: document.body }] });
            map.click({ target: document.body, ...eventCommonProps });
            jest.advanceTimersByTime(800);
        });
        wrapper.update();
        checkHasNoPopoverInner(wrapper);

        // should support the property `duration`
        wrapper.setProps({ duration: 1000 });
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);
        act(() => {
            jest.advanceTimersByTime(2000);
        });
        wrapper.update();
        checkHasNoPopoverInner(wrapper);
    });

    it('should adjust itself correctly when shown on the edge', () => {
        const ref = createRef();
        const wrapper = mount(<Popover
            content="Bubble"
            ref={ref}
            direction="bottomCenter"
        >
            <Button>Button</Button>
        </Popover>);
        clickButton(wrapper);
        mockElementProperty(screen, 'availWidth', 375);
        window.innerHeight = 700;
        const { child } = ref.current;
        expect(typeof child).toBe('object');
        // bottom overflow
        mockPopoverSize(ref, 700, 100);
        child.getBoundingClientRect = jest.fn(() => ({ top: 600, bottom: 690, left: 10, right: 100, width: 90, height: 90 }));
        updatePopoverPosition(wrapper, ref);
        expect(wrapper.find('.popover-content').hasClass('topCenter')).toBe(true);
        // top overflow
        mockPopoverSize(ref, 375, 100);
        child.getBoundingClientRect = jest.fn(() => ({ top: 10, bottom: 100, left: 10, right: 100, width: 90, height: 90 }));
        wrapper.setProps({ direction: 'topLeft' });
        wrapper.update();
        expect(wrapper.find('.popover-content').hasClass('bottomLeft')).toBe(true);
        wrapper.setProps({ mode: 'global' });
        expect(wrapper.find(`.${prefix}-inner`).hasClass('global-mode')).toBe(true);
    });

    it('should support Popover.Menu', () => {
        const onChange = jest.fn();
        const onSelect = jest.fn();
        const wrapper = mount(<Popover.Menu
            onChange={onChange}
            onSelect={onSelect}
            menu={[
                {
                    value: 'first',
                    text: 'Menu 1',
                },
                {
                    text: 'Menu 2',
                    disabled: true,
                },
                {
                    text: 'Menu 3',
                    icon: <div />,
                },
                'Menu 4',
            ]}
            theme="white"
            key="vertical"
            clickSelfToClose={false}
        >
            <Button>Button</Button>
        </Popover.Menu>);
        clickButton(wrapper);
        checkHasPopoverInner(wrapper);
        expect(wrapper.find(`.${prefix}-menu-content`)).toHaveLength(1);
        expect(wrapper.find(`.${prefix}-menu-item`)).toHaveLength(4);
        expect(wrapper.find(`.${prefix}-menu-item`).at(1).hasClass('disabled')).toBe(true);
        expect(wrapper.find(`.${prefix}-menu-item`).at(2).find(`.${prefix}-menu-icon`)).toHaveLength(1);
        // disabled item cannot trigger onSelect
        wrapper.find(`.${prefix}-menu-item`).at(1).simulate('click');
        expect(onSelect.mock.calls).toHaveLength(0);
        wrapper.find(`.${prefix}-menu-item`).at(0).simulate('touchstart');
        wrapper.find(`.${prefix}-menu-item`).at(0).simulate('click');
        expect(onSelect.mock.calls).toHaveLength(1);
        expect(onSelect.mock.calls[0]).toEqual(['first', {
            value: 'first',
            text: 'Menu 1',
        }]);
        expect(onChange.mock.calls).toHaveLength(1);
        wrapper.setProps({ useClickStatus: true, clickStatusDuration: 500, clickSelfToClose: true });
        wrapper.find(`.${prefix}-menu-item`).at(2).simulate('touchstart');
        wrapper.find(`.${prefix}-menu-item`).at(2).simulate('click');
        expect(wrapper.find(`.${prefix}-menu-item`).at(2).hasClass('active')).toBe(true);
        act(() => {
            jest.advanceTimersByTime(600);
        });
        wrapper.update();
        expect(wrapper.find(`.${prefix}-menu-item`)).toHaveLength(0);
        expect(onChange.mock.calls).toHaveLength(2);
        expect(onChange.mock.calls[1]).toEqual([false]);
    });
});
