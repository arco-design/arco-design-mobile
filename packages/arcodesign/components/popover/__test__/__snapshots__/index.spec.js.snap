// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover demo test popover demo: base.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="popover-base-demo-wrapper"
  >
    <div
      class="base-item"
    >
      <div
        class="arco-popover popover-with-margin white-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              White
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="base-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Black
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: custom.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="custom-demo-wrapper"
  >
    <div
      class="arco-popover custom-popover black-theme follow-mode"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Special bubble style
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: direction.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="direction-demo-wrapper"
  >
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Top left
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Top right
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Bottom left
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Bottom right
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Top center
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="direction-item"
    >
      <div
        class="arco-popover black-theme follow-mode"
      >
        <div
          class="popover-child-inner"
        >
          <button
            class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
            type="button"
          >
            <div
              class="arco-button-text arco-button-text-pc btn-text"
            >
              Bottom center
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: menu.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="menu-demo-wrapper"
  >
    <div
      class="arco-popover popover-with-margin arco-popover-menu white-theme follow-mode bordered custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            White
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover arco-popover-menu black-theme follow-mode custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Black
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: menu-with-click-status.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="menu-demo-wrapper"
  >
    <div
      class="arco-popover popover-with-margin arco-popover-menu white-theme follow-mode bordered custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            White
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover arco-popover-menu black-theme follow-mode custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Black
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: menu-with-disabled-status.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="menu-demo-wrapper"
  >
    <div
      class="arco-popover popover-with-margin arco-popover-menu white-theme follow-mode bordered custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            White
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover arco-popover-menu black-theme follow-mode custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Black
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: menu-with-icon.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="menu-demo-wrapper"
  >
    <div
      class="arco-popover popover-with-margin arco-popover-menu white-theme follow-mode bordered custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          style="padding: 0px;"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Vertical menu
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover arco-popover-menu black-theme follow-mode custom-content"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          style="padding: 0px;"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Horizontal menu
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: suffix.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="suffix-demo-wrapper"
  >
    <div
      class="arco-popover popover-with-margin black-theme follow-mode"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Close Action
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover black-theme follow-mode"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Button Action
          </div>
        </button>
      </div>
    </div>
    <div
      class="arco-popover black-theme follow-mode"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            Text Action
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`popover demo test popover demo: white-theme-mask.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="white-theme-mask-demo-wrapper"
  >
    <div
      class="arco-popover white-theme global-mode bordered"
    >
      <div
        class="popover-child-inner"
      >
        <button
          class="arco-button arco-button-type-primary type-primary arco-button-size-large arco-button-size-large-is-semi size-large is-semi pc"
          type="button"
        >
          <div
            class="arco-button-text arco-button-text-pc btn-text"
          >
            White bubble with mask
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
